sszgen/generator/unmarshal.go

```diff
@@ -39,22 +39,18 @@ func (v *Value) unmarshal(dst string) string {
 			validate = fmt.Sprintf("if len(%s) > %d { return ssz.ErrBytesLength }\n", dst, v.m)
 		}
 
-		refName := ""
-		if v.ref != "" {
-			refName = v.ref + "." + v.obj
-		}
-
 		// both fixed and dynamic are decoded equally
 		tmpl := `{{.validate}}if cap(::.{{.name}}) == 0 {
-			{{if .refName}} ::.{{.name}} = {{ .refName }}(make([]byte, 0, len({{.dst}}))) {{ else }} ::.{{.name}} = make([]byte, 0, len({{.dst}})) {{ end }}
+			{{if .isRef}} ::.{{.name}} = {{ ref .obj }}(make([]byte, 0, len({{.dst}}))) {{ else }} ::.{{.name}} = make([]byte, 0, len({{.dst}})) {{ end }}
```

Comment:
The issue was here, you have to use the new `ref` function during the template to record the values that need to be imported.